### PR TITLE
[rllib] Make Pong-v0 + EvolutionStrategies work by sharing preprocessors with PPO

### PR DESCRIPTION
**Pull request by ericl**
https://github.com/ray-project/ray/issues/847

The issue was basically that ES did not preprocess its observations. This fixes the issue by sharing preprocessors between policy gradient and ES. In the future, we should port other algorithms to also use the same preprocessors and filters.

Tested that ES now runs with both CartPole-v0 and Pong-v0.

### CODE REVIEW

File: python/ray/rllib/models/preprocessors.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class AtariPixelPreprocessor(Preprocessor):
+    def transform_shape(self, obs_shape):
+        return (80, 80, 3)
+
+    # TODO(ekl) why does this need to return an extra size-1 dim (the [None])
+    def transform(self, observation):
+        """Downsamples images from (210, 160, 3) to (80, 80, 3)."""
+        return (observation[25:-25:2, ::2, :][None] - 128) / 128
```
Comment (ericl) [inline python/ray/rllib/models/preprocessors.py]: @pcmoritz do you know why the `[None]` is needed here? This causes the shape to be `(1, 80, 80, 3)` which trips up ES unless you squeeze out the extra dim.

However PPO seems to be happy with the extra dim, and in fact it broke when I tried getting rid of it:

```
Traceback (most recent call last):
  File "/home/eric/Desktop/ray-private/python/ray/worker.py", line 732, in _process_task
    self.actors[task.actor_id().id()], *arguments)
  File "/home/eric/Desktop/ray-private/python/ray/rllib/policy_gradient/agent.py", line 194, in compute_steps
    for key, val in trajectory.items()}
  File "/home/eric/Desktop/ray-private/python/ray/rllib/policy_gradient/agent.py", line 194, in <dictcomp>
    for key, val in trajectory.items()}
IndexError: boolean index did not match indexed array along dimension 0; dimension is 87120 but corresponding boolean dimension is 1089
```

Didn't investigate further.

File: python/ray/rllib/evolution_strategies/evolution_strategies.py [hunk 2]
```diff
@@ -30,7 +31,7 @@
 DEFAULT_CONFIG = dict(
     l2coeff=0.005,
     noise_stdev=0.02,
-    episodes_per_batch=1000,
```
Comment (pcmoritz) [inline python/ray/rllib/evolution_strategies/evolution_strategies.py]: shall we restore this back?

**Comment by pcmoritz** [comment]
LGTM, let's fix this and address the extra dimension in a separate PR

**Comment by robertnishihara** [comment]
Note that this seems to have broken the evolution strategies test on Jenkins.

**Comment by ericl** [comment]
Updated. The problem was that the code for picking the preprocessor wasn't very robust and it picked `AtariPixelPreprocessor` for the pendulum env. We can fix that as a follow-up though.

**Pull request merged**